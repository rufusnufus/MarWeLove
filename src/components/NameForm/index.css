.name-form {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  display: flex;
  flex-direction: column;
  max-width: 600px;
  padding: 40px;
  background-color: #ec1d24;
  border-radius: 5px;
  font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
  font-size: 30px;
  text-align: center;
  color: black;
  align-items: center;
  z-index: 9999;
}

.name-form input {
  margin-top: 20px;
  border: none;
  padding: 10px;
  border-radius: 5px;
  font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
  min-width: 60%;
  font-size: 20px;
}

.name-form span {
  color: white;
}

.name-form button {
  margin-top: 20px;
  font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
  font-size: 30px;

  color: white;
  background-color: black;
  border: none;
  border-radius: 5px;
  min-width: 60%;
  cursor: pointer;
}

.name-form button:hover {
  scale: 1.05;
}
