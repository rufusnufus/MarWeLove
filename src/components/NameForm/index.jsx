import React, { useState } from 'react'
import { useFormik } from 'formik'
import { useDispatch } from 'react-redux'
import { setName, setToken } from '../../store/slices/user'

import './index.css'
import apiService from '../../services/apiService'

const NameForm = () => {
  const dispatch = useDispatch()
  const [status, setStatus] = useState('')

  const formik = useFormik({
    initialValues: {
      name: '',
      password: ''
    },
    onSubmit: (values) => {
      const { name, password } = values
      apiService
        .performLogin(name, password)
        .then((results) => {
          dispatch(setName(values.name))
          dispatch(setToken(results))
        })
        .catch(() => {
          setStatus('ERROR')
        })
    }
  })

  function handleRegister(values) {
    const { name, password } = values
    apiService
      .performRegister(name, password)
      .then(() => {
        setStatus('OK')
      })
      .catch(() => {
        setStatus('ERROR')
      })
  }

  return (
    <form className="name-form" onSubmit={formik.handleSubmit}>
      <label htmlFor="name">
        What`s your name, <span>hero</span>?
      </label>

      <input id="name" name="name" type="text" onChange={formik.handleChange} value={formik.values.name} />
      <input
        id="password"
        name="password"
        type="password"
        onChange={formik.handleChange}
        value={formik.values.password}
      />

      <button onClick={() => handleRegister(formik.values)} type="button">
        Register
      </button>
      <button type="submit">Login</button>
      {status === 'OK' && <p> Registration completed </p>}
      {status === 'ERROR' && <p> Request failed </p>}
    </form>
  )
}

export default NameForm
