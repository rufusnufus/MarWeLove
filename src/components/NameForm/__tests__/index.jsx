/**
 * @jest-environment jsdom
 */

import React from 'react'
import { cleanup, render } from '@testing-library/react'
import { Provider } from 'react-redux'
import NameForm from '../index'
import mockStore from '../../../__mocks__/store'

let store
beforeEach(() => {
  store = mockStore()
})
afterEach(cleanup)

describe('NameForm', () => {
  it('Render', () => {
    const { container } = render(
      <Provider store={store}>
        <NameForm />
      </Provider>
    )

    expect(container).toMatchSnapshot()
  })
})
