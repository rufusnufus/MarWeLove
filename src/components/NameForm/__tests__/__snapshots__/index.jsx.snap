// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameForm Render 1`] = `
<div>
  <form
    class="name-form"
  >
    <label
      for="name"
    >
      What\`s your name, 
      <span>
        hero
      </span>
      ?
    </label>
    <input
      id="name"
      name="name"
      type="text"
      value=""
    />
    <input
      id="password"
      name="password"
      type="password"
      value=""
    />
    <button
      type="button"
    >
      Register
    </button>
    <button
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
